seshat-node/lib/index.js

```diff
@@ -393,4 +445,64 @@ class Seshat extends seshat.Seshat {
     }
 }
 
-module.exports = Seshat;
+/**
+ * Seshat recovery database.<br>
+ *
+ * A Seshat recovery database can be used to re-index a Seshat database.
+ *
+ * This will be needed if schema changes to the index were required and the
+ * library has been upgraded.
+ *
+ * The recovery database uses the same parameters in the constructor like the
+ * normal Seshat database.
+ *
+ * @param {string} path The path where the database should be stored. If a
+ * database already exist in the given folder the database will be reused.
+ * @param {object} config Additional configuration for the database.
+ * database already exist in the given folder the database will be reused.
+ * @param  {string} config.language The language that the database should use
+ * for indexing. Picking the correct indexing language may improve the search.
+ * @param  {string} config.passphrase The passphrase that should be used to
+ * encrypt the database. The database is left unencrypted it no passphrase is
+ * set.
+ *
+ * @constructor
+ *
+ * @example
+ * // open a Seshat recovery database in the given folder
+ * let recovery = new SeshatRecovery("/home/example/database_dir");
+ * // reindex the database
+ * await recovery.reindex();
+ */
+class SeshatRecovery extends seshat.SeshatRecovery {
```

Comment:
I might have feedback after seeing how you plan to handle the recovery path in Riot, but at an abstract level it seems okay.

Reply:
This commit implements support for it in Riot: https://github.com/vector-im/riot-web/commit/921d734ef65de26f5a58394beee4ac001ee87085.

I didn't notice any delays at startup, the thing seems to be reasonably quick but do note that this was using a 3k events large database.